Source/FakeItEasy/Core/FakeScope.cs

```diff
@@ -13,19 +16,48 @@ namespace FakeItEasy.Core
     internal abstract class FakeScope
         : IFakeScope
     {
+        private const string CurrentScopeLogicalCallContextKey = "FakeItEasy.FakeScope.Current";
+
+        private static readonly RootScope root;
+        private static readonly ConcurrentDictionary<Guid, FakeScope> scopes;

```

Comment:
Why, on the context, pass GUIDs that point to fakescopes inside a dictionary instead of just passing object pointers to the fakescopes?